V2 ImportState Testing Ignoring ExpectError

### SDK version

```
v2.0.1
```

### Relevant provider source code

```go
func TestAccAWSKinesisFirehoseDeliveryStream_basic(t *testing.T) {
	resourceName := "aws_kinesis_firehose_delivery_stream.test"
	rInt := acctest.RandInt()

	funcName := fmt.Sprintf("aws_kinesis_firehose_ds_import_%d", rInt)
	policyName := fmt.Sprintf("tf_acc_policy_%d", rInt)
	roleName := fmt.Sprintf("tf_acc_role_%d", rInt)
	var stream firehose.DeliveryStreamDescription

	config := testAccFirehoseAWSLambdaConfigBasic(funcName, policyName, roleName) +
		fmt.Sprintf(testAccKinesisFirehoseDeliveryStreamConfig_extendedS3basic,
			rInt, rInt, rInt, rInt)

	resource.ParallelTest(t, resource.TestCase{
		PreCheck:     func() { testAccPreCheck(t) },
		Providers:    testAccProviders,
		CheckDestroy: testAccCheckKinesisFirehoseDeliveryStreamDestroy,
		Steps: []resource.TestStep{
			{
				Config: config,
				Check: resource.ComposeTestCheckFunc(
					testAccCheckKinesisFirehoseDeliveryStreamExists(resourceName, &stream),
					testAccCheckAWSKinesisFirehoseDeliveryStreamAttributes(&stream, nil, nil, nil, nil, nil),
				),
			},
			{
				ResourceName:      resourceName,
				ImportState:       true,
				ImportStateVerify: true,
			},
			// Ensure we properly error on malformed import IDs
			{
				ResourceName:  resourceName,
				ImportState:   true,
				ImportStateId: "just-a-name",
				ExpectError:   regexp.MustCompile(`Expected ID in format`),
			},
			{
				ResourceName:  resourceName,
				ImportState:   true,
				ImportStateId: "arn:aws:firehose:us-east-1:123456789012:missing-slash",
				ExpectError:   regexp.MustCompile(`Expected ID in format`),
			},
		},
	})
}
```

### Terraform Configuration Files

Please see the Terraform AWS Provider repository, but presumably this would happen with any resource configuration and `ImportState` + `ExpectError` defined.

### Debug Output

Please ask if you need it.

### Expected Behavior

`ExpectError` passes when import error message matches regular expression, which was the behavior in SDK v1. Otherwise this change to be documented in the SDK v2 upgrade guide.

### Actual Behavior

```
=== CONT  TestAccAWSKinesisFirehoseDeliveryStream_basic
    TestAccAWSKinesisFirehoseDeliveryStream_basic: testing_new_import_state.go:74: failed to import: terraform failed: exit status 1

        stderr:

        Error: Expected ID in format of arn:PARTITION:firehose:REGION:ACCOUNTID:deliverystream/NAME and provided: just-a-name



--- FAIL: TestAccAWSKinesisFirehoseDeliveryStream_basic (101.43s)
```

### Steps to Reproduce

1. `TF_ACC=1 go test ./aws -v -count 1 -timeout 120m -parallel 20 -run='TestAccAWSKinesisFirehoseDeliveryStream_basic'`

--- a/helper/resource/testing_new.go
+++ b/helper/resource/testing_new.go
@@ -111,7 +111,7 @@ func runNewTest(t testing.T, c TestCase, helper *tftest.Helper) {
 					t.Fatalf("Step %d/%d error running import: expected an error but got none", i+1, len(c.Steps))
 				}
 				if !step.ExpectError.MatchString(err.Error()) {
-					t.Fatalf("Step %d/%d error running import, expected an error with pattern, no match on: %s", i+1, len(c.Steps), err)
+					t.Fatalf("Step %d/%d error running import, expected an error with pattern (%s), no match on: %s", i+1, len(c.Steps), step.ExpectError.String(), err)
 				}
 			} else {
 				if err != nil {

--- a/helper/resource/testing_new.go
+++ b/helper/resource/testing_new.go
@@ -106,8 +106,17 @@ func runNewTest(t testing.T, c TestCase, helper *tftest.Helper) {
 
 		if step.ImportState {
 			err := testStepNewImportState(t, c, helper, wd, step, appliedCfg)
-			if err != nil {
-				t.Fatal(err)
+			if step.ExpectError != nil {
+				if err == nil {
+					t.Fatalf("Step %d/%d error running import: expected an error but got none", i+1, len(c.Steps))
+				}
+				if !step.ExpectError.MatchString(err.Error()) {
+					t.Fatalf("Step %d/%d error running import, expected an error with pattern, no match on: %s", i+1, len(c.Steps), err)
+				}
+			} else {
+				if err != nil {
+					t.Fatalf("Step %d/%d error running import: %s", i+1, len(c.Steps), err)
+				}
 			}
 			continue
 		}

--- a/helper/resource/testing_new_import_state.go
+++ b/helper/resource/testing_new_import_state.go
@@ -71,11 +71,10 @@ func testStepNewImportState(t testing.T, c TestCase, helper *tftest.Helper, wd *
 	}
 
 	err = runProviderCommand(t, func() error {
-		importWd.RequireImport(t, step.ResourceName, importId)
-		return nil
+		return importWd.Import(step.ResourceName, importId)
 	}, importWd, c.ProviderFactories)
 	if err != nil {
-		t.Fatalf("Error running import: %s", err)
+		return err
 	}
 
 	var importState *terraform.State

